cmd/start.go

```diff
@@ -0,0 +1,60 @@
+package cmd
+
+import (
+	"os/signal"
+	"syscall"
+
+	"github.com/spf13/cobra"
+
+	"github.com/celestiaorg/celestia-node/node"
+)
+
+// Start constructs a CLI command to start Celestia Node daemon of the given type 'tp'.
+// It is meant to be used a subcommand and also receive persistent flag name for repository path.
+func Start(repoName string, tp node.Type) *cobra.Command {
+	if !tp.IsValid() {
+		panic("cmd: Start: invalid Node Type")
+	}
+	if len(repoName) == 0 {
+		panic("parent command must specify a persistent flag name for repository path")
+	}
+	return &cobra.Command{
+		Use:          "start",
+		Short:        "Starts Node daemon. First stopping signal gracefully stops the Node and second terminates it.",
+		Aliases:      []string{"run", "daemon"},
+		Args:         cobra.NoArgs,
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			repoPath := cmd.Flag(repoName).Value.String()
+
+			repo, err := node.Open(repoPath, tp)
+			if err != nil {
+				return err
+			}
+
+			nd, err := node.New(tp, repo)
+			if err != nil {
+				return err
+			}
+
+			ctx, cancel := signal.NotifyContext(cmd.Context(), syscall.SIGINT, syscall.SIGTERM)
+			defer cancel()
+			err = nd.Start(ctx)
+			if err != nil {
+				return err
+			}
+
+			<-ctx.Done()
+			cancel() // ensure we stop reading more signals for start context
+
+			ctx, cancel = signal.NotifyContext(cmd.Context(), syscall.SIGINT, syscall.SIGTERM)
+			defer cancel()
+			err = nd.Stop(ctx)
+			if err != nil {
+				return err
+			}
+
+			return repo.Close()
```

Comment:
`repo.Close()` should also be called inside `nd.Stop()`, there shouldn't be a circumstance where you want to stop the node but leave the fslock around.

Reply:
Well, there is a trade-off here. The Node depends on Repository from API pov and giving dependable component control over the lifecycle is not a good practise. It is like giving an ‘os.File’ to ‘ReadFull’ and making it close the File.

Reply:
I would say at the very least provide a pair of functions (for example) "node.Begin" and "node.End" that do both and handle the "repo" open/close stuff internally. If you want them to still be able to take parameters for different types of repos then maybe create a separate "repo" interface for which an object's implementation can be fed in as an argument, and the `node` package will store that object and use it to open/close.

Though that might be a bit out of scope for this PR.

Reply:
Repository is an interface...
Pls, open an issue if you see a problem here. Indeed, that is out of the scope of the PR

Reply:
Right I see that it's an interface, got confused with something else. Looks good for now then.

Reply:
Referencing #30 as the name keeps bubbling up. I don't think it is a high priority to change it though.

Reply:
https://github.com/celestiaorg/celestia-node/issues/146